13_biblioteki/13_8_AlgorithmsBLAS.tex

```diff
@@ -0,0 +1,257 @@
+% !TEX encoding = UTF-8 Unicode
+\section{Algorytmy blokowe i BLAS}
+%%%%%%%%%%%%%%%%
+	\begin{frame}{Faktoryzacja}
+		Level 1, 2, 3 BLAS performance \\
+		Blocked algorithms \\
+		$$
+		\begin{pmatrix}
+		A_{11} & a_j & A_{13} \\
+		. & a_{jj} & a_j^T \\
+		. & . & A_{33}
+		\end{pmatrix}
+		=
+		\begin{pmatrix}
+		U_{11}^T & 0 & 0 \\
+		u_j^T & u_{jj} & 0 \\
+		U_{13}^T & u_j & U_{33}^T
+		\end{pmatrix}
+		*
+		\begin{pmatrix}
+		U_{11} & u_j  & U_{13} \\
+		0 & u_{jj} & u_j^T \\
+		0 & 0 & U_{33}
+		\end{pmatrix}
+		$$
+		Porównując współczynniki w kolumnie j otrzymujemy:
+		$
+		\begin{cases}
+		a_j = U_{jj}^T * u_j \\
+		a_{jj} = u_j^T * u_j + u_{jj}^2
+		\end{cases}
+		$ \\
+		Jeżeli $U_{11}$ został wyznaczony, to $u_j$ oraz $u_{jj}$ można obliczyć z:
+		$
+		\begin{cases}
+		U_{11}^T * u_j = a_j \\
+		u_{jj}^T = a_jj - u_j^T * u_{j}
+		\end{cases}
+		$
+	\end{frame}
+	\begin{frame}[fragile]{Realizacja z użyciem BLAS level 1}
+		\begin{lstlisting}
+			DO 30 J = 1, N
+				INFO = J
+				S = 0.0E0
+				JM1 = J - 1
+				IF(JM1 .LT. 1) GO TO 20
+				DO 10 K = 1, JM1
+					T = A(K,J) - SDOT(K-1, A(1,K), 1,
+					A(1,J), 1)
+					T = T/A(K,K)
+					A(K,J) = T
+					S = S + T*T
+			10 CONTINUE
+			20 CONTINUE
+			S = A(J,J) - S
+			... EXIT
+			IF (S .LE. 0.0E0) GO TO 40 A(J,J) = SQRT(S)
+			30 CONTINUE		
+		\end{lstlisting}
+	\end{frame}
+	\begin{frame}{Realizacja z użyciem BLAS level 1 I}
+		\begin{tabular}{ l l l }
+			\textbf{FUNCTION} & \textunderscore \textbf{DOT(N,X, INCX,Y,INCY) = dot} & $ \leftarrow x^T * y $ \\
+			   & $\uparrow$ & $ \leftarrow x^H * y $ \\
+			   & S &  \\
+			   & D & \\
+			   & DS & \\
+		\end{tabular} \\ 
+		\textbf{N} - number of elements of vectores x and y to be used in dot \\
+		\textbf{X} - array of length $lenx = (n - 1) * |inc x| + 1$ containing the n-vector $x$ \\
+		\textbf{INCX} - increment for the array $x$: \\
+		$\textit{incx} \geq 0 $ $x$ stored forward in array $x$, i.e: $x_i \rightarrow x((i - 1) * incx + 1)$ \\
+		$\textit{incx} < 0$ $x$ stored backward in array $x$, i.e: $x_i \rightarrow x((i - n) * incx + 1)$ \\
+		\textbf{Y} - \\
+		\textbf{INCY} - \\
+	\end{frame}
+	\begin{frame}{Realizacja z użyciem BLAS level 1 II}
+		OUTPUT: $s = \sum_{i=1}^n x_i * y_i$, $s = \sum_{i=1}^n x_i^{*} * y_i$ \\
```

Comment:
Mybe insert a newline after 'OUTPUT' and put all the stuff below in the center? I think it'd be more readable that way.